main.py

```diff
@@ -0,0 +1,117 @@
+from datetime import datetime
+"""
+open all 3 files
+parse such that we have the log in a list
+count how many lines are in log, level=error count, level=info
+
+TIME FORMAT : 08:56:35.354
+
+top 5 most requested HTTP paths/endpoints
+"""
+
+class Logging:
+    def __init__(self):
+        self.log = []
+        self.date_time = []
+
+
+    def logs(self, file1, file2, file3):
+        file_1 = open(file1, 'r')
+        file_2 = open(file2, 'r')
+        file_3 = open(file3, 'r')
+
+        for line in file_1:
+            self.log.append(line)
+        for line in file_2:
+            self.log.append(line)
+        for line in file_3:
+            self.log.append(line)
+
+        file_1.close()
+        file_2.close()
+        file_3.close()
+
+    def get_entry_count(self):
+        return len(self.log)
+
+    def get_level_count(self, level):
+        count = 0
+
+        for line in self.log:
+            if level in line:
+                count += 1
+
+        return count
+
+    def log_times(self):
+        #TESTCASE: self.date_time.append(datetime.strptime(self.log[0].split("\"", 2)[1], '%Y-%m-%d %H:%M:%S.%f'))
+
+        #DEBUG: for count, val in enumerate(self.log):
+            #print(count)
+            #print(val.split("\"", 2)[1])
+
+        for i in range(len(self.log)):
+            if self.log[0].split("\"", 2)[0] in self.log[i].split("\"", 2)[0]:
```

Comment:
Two thoughts here:

1. Again not a huge ding because if you were doing this under interview time constraints I wouldn't expect this to be perfect, but if you're gonna create a test case, you can include a few lines of the actual log file and then prove that those parse correctly. This would be how you would unit test this.
2. Because I don't know what the log actually looks like, I have no idea what is happening here. If the code you write ever looks this gnarly, it's a good suggestion that you probably should throw in some comments to explain what the hell is going on. If the interview goes back and can't explain what happened in the interview review then it's gonna not be so good for you.